File: RepoTransBench/source_projects/Python/scrapinghub_adblockparser/public_tests/test_public_options.py
import pytest
from adblockparser import AdblockRule

def test_public_parse_options_simple():
    r = AdblockRule('/ad.js$script,domain=example.com|another.net')
    assert 'script' in r.options
    assert 'domain' in r.options
    assert r.options['domain'] == {'example.com', 'another.net'}

def test_public_parse_options_no_options():
    r = AdblockRule('/track.gif')
    assert r.options == {}

def test_public_parse_options_complex():
    r = AdblockRule('/analytics.js$image,third-party,domain=mysite.org|anothersite.co.uk')
    assert 'image' in r.options
    assert 'third-party' in r.options
    assert r.options['domain'] == {'mysite.org', 'anothersite.co.uk'}

def test_public_parse_options_exception():
    r = AdblockRule('@@/nocache.png$subdocument,domain=sub.example.org')
    assert r.is_exception
    assert 'subdocument' in r.options
    assert r.options['domain'] == {'sub.example.org'}

File: RepoTransBench/target_projects/Python/Java/scrapinghub_adblockparser/src/test/java/com/scrapinghub/adblockparser/publictests/TestPublicOptions.java
package com.scrapinghub.adblockparser.publictests;

import com.scrapinghub.adblockparser.AdblockRule;
import org.junit.jupiter.api.Test;

import java.util.*;

import static org.junit.jupiter.api.Assertions.*;

public class TestPublicOptions {
    @Test
    public void testPublicParseOptionsSimple() {
        AdblockRule r = new AdblockRule("/ad.js$script,domain=example.com|another.net");
        assertTrue(r.options.containsKey("script"));
        assertTrue(r.options.containsKey("domain"));
        assertEquals(new HashSet<>(Arrays.asList("example.com", "another.net")), r.options.get("domain"));
    }

    @Test
    public void testPublicParseOptionsNoOptions() {
        AdblockRule r = new AdblockRule("/track.gif");
        assertEquals(Collections.emptyMap(), r.options);
    }

    @Test
    public void testPublicParseOptionsComplex() {
        AdblockRule r = new AdblockRule("/analytics.js$image,third-party,domain=mysite.org|anothersite.co.uk");
        assertTrue(r.options.containsKey("image"));
        assertTrue(r.options.containsKey("third-party"));
        assertEquals(new HashSet<>(Arrays.asList("mysite.org", "anothersite.co.uk")), r.options.get("domain"));
    }

    @Test
    public void testPublicParseOptionsException() {
        AdblockRule r = new AdblockRule("@@/nocache.png$subdocument,domain=sub.example.org");
        assertTrue(r.isException);
        assertTrue(r.options.containsKey("subdocument"));
        assertEquals(new HashSet<>(Arrays.asList("sub.example.org")), r.options.get("domain"));
    }
}